### PR TITLE
initial commit of CountJPGs.py

### PR DESCRIPTION
**Pull request by bob-skowron**
For problem 3

### CODE REVIEW

**Comment by jasonwalker80** [comment]
I can only get this to run using the default value of `--deploy-mode`  which is client `client`. I can *NOT* get this to run using `--master local`. I must be missing something.

Using local disk:
```
hdfs dfs -get /loudacre/weblogs/*
pyspark CountJPGs.py file:///home/training/countjpgs/FlumeData.*
```

Using HDFS
```
pyspark CountJPGs.py /loudacre/weblogs/FlumeData.*
```

The output is:
```
jpgs/CountJPGs.py:15, took 27.912800 s
64978
```

I can't figure out the syntax to run locally or to run using local files (not hdfs). I tried using the `--py-files` option and the `--files` option with pyspark, but I could not get it to work. Also, the spark job doesn't exit when run without an input argv. Maybe that should be a

File: hw9/CountJPGs.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import sys
+from pyspark import SparkContext
+
+if __name__ == "__main__":
+	if len(sys.argv) < 1:
```
Comment (jasonwalker80) [inline hw9/CountJPGs.py:5]: I think this should be `!= 2`. With python `argv[0]` is the command and `argv[1]` is the path to the file(s).

**Review by jasonwalker80 — COMMENTED** [review]
I think this works, but I'm not sure if we need to use SparkConf. How do we tell pyspark to run local? I'm pretty sure we can use `--driver-mode cluster` for part E, but I can't figure out how to run locally for part B of this problem.

**Comment by jasonwalker80** [comment]
I see your examples using spark-submit (appears to be the same as pyspark). Is the default to run locally? Can you combine  `-master yarn-client` instead of defining `-master yarn` *AND* `-deploy-mode client`? Reading the docs I would have thought `-master local` would work, but it doesn't for me. I guess ignore the above comments if your solution in the PDF and tex files works. I confirmed the answer.

**Comment by jasonwalker80** [comment]
I ran with your args and it all worked. If you agree with the `!= 2` comment then we should update the PR and I can submit the python script. This one has to go in our SVN.

File: hw9/CountJPGs.py (hunk 1 above)
Comment (bob-skowron) [reply hw9/CountJPGs.py:5]: concur

**Pull request merged**